### ISSUE TITLE
Sending mails broken

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/mail#1091]
Trying to reply to an email, I get the »Server error« notification. However my owncloud.log says:

```
{"reqId":"AoVsVf5mji7TD+DLQBbV","remoteAddr":"127.0.0.1","app":"index","message":"Exception: {\"Exception\":\"OCP\\\\AppFramework\\\\Db\\\\DoesNotExistException\",\"Message\":\"No matching entry found\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/public\\\/appframework\\\/db\\\/mapper.php(346): OCP\\\\AppFramework\\\\Db\\\\Mapper->findOneQuery('SELECT * FROM `...', Array, NULL, NULL)\\n#1 \\\/home\\\/jan\\\/owncloud\\\/apps\\\/mail\\\/lib\\\/db\\\/mailaccountmapper.php(47): OCP\\\\AppFramework\\\\Db\\\\Mapper->findEntity('SELECT * FROM `...', Array)\\n#2 \\\/home\\\/jan\\\/owncloud\\\/apps\\\/mail\\\/lib\\\/service\\\/accountservice.php(51): OCA\\\\Mail\\\\Db\\\\MailAccountMapper->find('jan', 0)\\n#3 \\\/home\\\/jan\\\/owncloud\\\/apps\\\/mail\\\/lib\\\/controller\\\/accountscontroller.php(261): OCA\\\\Mail\\\\Service\\\\AccountService->find('jan', 0)\\n#4 [internal function]: OCA\\\\Mail\\\\Controller\\\\AccountsController->send(0, '', NULL, 'Yeah, cool stuf...', '\\\"Jos ..', '\\\"JeanLuc\\\" ..', '', 0, '', NULL)\\n#5 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(159): call_user_func_array(Array, Array)\\n#6 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Mail\\\\Controller\\\\AccountsController), 'send')\\n#7 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/app.php(108): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Mail\\\\Controller\\\\AccountsController), 'send')\\n#8 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/routing\\\/routeactionhandler.php(45): OC\\\\AppFramework\\\\App::main('AccountsControl...', 'send', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#9 [internal function]: OC\\\\AppFramework\\\\routing\\\\RouteActionHandler->__invoke(Array)\\n#10 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(281): call_user_func(Object(OC\\\\AppFramework\\\\routing\\\\RouteActionHandler), Array)\\n#11 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/base.php(893): OC\\\\Route\\\\Router->match('\\\/apps\\\/mail\\\/acco...')\\n#12 \\\/home\\\/jan\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#13 {main}\",\"File\":\"\\\/home\\\/jan\\\/owncloud\\\/lib\\\/public\\\/appframework\\\/db\\\/mapper.php\",\"Line\":284}","level":4,"time":"2015-09-16T14:56:26+00:00","method":"POST","url":"\/owncloud\/index.php\/apps\/mail\/accounts\/%7BaccountId%7D\/send"}
{"reqId":"\/qLYeraa8iY7s37tvZW5","remoteAddr":"127.0.0.1","app":"mail","message":"Updating draft <0> in account <0>","level":1,"time":"2015-09-16T14:56:28+00:00","method":"POST","url":"\/owncloud\/index.php\/apps\/mail\/accounts\/%7BaccountId%7D\/draft"}
{"reqId":"\/qLYeraa8iY7s37tvZW5","remoteAddr":"127.0.0.1","app":"index","message":"Exception: {\"Exception\":\"OCP\\\\AppFramework\\\\Db\\\\DoesNotExistException\",\"Message\":\"No matching entry found\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/public\\\/appframework\\\/db\\\/mapper.php(346): OCP\\\\AppFramework\\\\Db\\\\Mapper->findOneQuery('SELECT * FROM `...', Array, NULL, NULL)\\n#1 \\\/home\\\/jan\\\/owncloud\\\/apps\\\/mail\\\/lib\\\/db\\\/mailaccountmapper.php(47): OCP\\\\AppFramework\\\\Db\\\\Mapper->findEntity('SELECT * FROM `...', Array)\\n#2 \\\/home\\\/jan\\\/owncloud\\\/apps\\\/mail\\\/lib\\\/service\\\/accountservice.php(51): OCA\\\\Mail\\\\Db\\\\MailAccountMapper->find('jan', 0)\\n#3 \\\/home\\\/jan\\\/owncloud\\\/apps\\\/mail\\\/lib\\\/controller\\\/accountscontroller.php(391): OCA\\\\Mail\\\\Service\\\\AccountService->find('jan', 0)\\n#4 [internal function]: OCA\\\\Mail\\\\Controller\\\\AccountsController->draft(0, NULL, 'Yeah, cool stuf...', '\\\"Jos ..', '\\\"JeanLuc\\\" ...', '', 0, '')\\n#5 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(159): call_user_func_array(Array, Array)\\n#6 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/http\\\/dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Mail\\\\Controller\\\\AccountsController), 'draft')\\n#7 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/app.php(108): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Mail\\\\Controller\\\\AccountsController), 'draft')\\n#8 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/appframework\\\/routing\\\/routeactionhandler.php(45): OC\\\\AppFramework\\\\App::main('AccountsControl...', 'draft', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#9 [internal function]: OC\\\\AppFramework\\\\routing\\\\RouteActionHandler->__invoke(Array)\\n#10 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(281): call_user_func(Object(OC\\\\AppFramework\\\\routing\\\\RouteActionHandler), Array)\\n#11 \\\/home\\\/jan\\\/owncloud\\\/lib\\\/base.php(893): OC\\\\Route\\\\Router->match('\\\/apps\\\/mail\\\/acco...')\\n#12 \\\/home\\\/jan\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#13 {main}\",\"File\":\"\\\/home\\\/jan\\\/owncloud\\\/lib\\\/public\\\/appframework\\\/db\\\/mapper.php\",\"Line\":284}","level":4,"time":"2015-09-16T14:56:28+00:00","method":"POST","url":"\/owncloud\/index.php\/apps\/mail\/accounts\/%7BaccountId%7D\/draft"}
```

@ChristophWurst @Gomez any idea?

**Comment by enoch85** [comment]
I can confirm this issue.

**Comment by jancborchardt** [comment]
@Gomez @ChristophWurst we need firefighting here! :D Don’t we have tests for this stuff? Core functionality like sending emails should be covered by autotests.

**Comment by itheiss** [comment]
I rember seen this exception working on #921. Maybe related to my dirty hack. Does this only happen from unified inbox?

**Comment by ChristophWurst** [comment]
> @Gomez @ChristophWurst we need firefighting here! :D Don’t we have tests for this stuff? Core functionality like sending emails should be covered by autotests.

Let me guess: it's an unified inbox issue again. The server probably tries to load the account from the DB (account ID = -1) and fails…

**Comment by ChristophWurst** [comment]
> Core functionality like sending emails should be covered by autotests.

If only someone was already working on bringing the code coverage up… #1078 #1080 #1082. Maybe you can make somebody review those at the meetup :P

**Comment by enoch85** [comment]
I use unified inbox, have not tested without.

**Comment by jancborchardt** [comment]
@ChristophWurst right, yep, it’s unified inbox. Works with single inbox. (I basically only use unified inbox cause it’s damn awesome!)

**Comment by jancborchardt** [comment]
@ChristophWurst ok, @Gomez @jakobsack will have to take a look at those test PRs. :)

**Comment by Gomez** [comment]
Fixed with #1097
